tokio/src/sync/mpsc/bounded.rs

```diff
@@ -530,6 +530,74 @@ impl<T> Receiver<T> {
         self.chan.len()
     }
 
+    /// Returns the current capacity of the channel.
+    ///
+    /// The capacity goes down when the sender sends a value by calling [`Sender::send`] or by reserving
+    /// capacity with [`Sender::reserve`]. The capacity goes up when values are received.
+    /// This is distinct from [`max_capacity`], which always returns buffer capacity initially
+    /// specified when calling [`channel`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::sync::mpsc;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let (tx, mut rx) = mpsc::channel::<()>(5);
+    ///
+    ///     assert_eq!(rx.capacity(), 5);
+    ///
+    ///     // Making a reservation drops the capacity by one.
+    ///     let permit = tx.reserve().await.unwrap();
+    ///     assert_eq!(rx.capacity(), 4);
+    ///
+    ///     // Sending and receiving a value increases the capacity by one.
+    ///     permit.send(());
+    ///     rx.recv().await.unwrap();
+    ///     assert_eq!(rx.capacity(), 5);
+    /// }
+    /// ```
+    /// [`capacity`]: Receiver::capacity
+    /// [`max_capacity`]: Receiver::max_capacity
+    pub fn capacity(&self) -> usize {
+        self.chan.semaphore().semaphore.available_permits()
+    }
```

Comment:
Hmm, this is not actually equal to `max_capacity - len`. Generally, sending a message goes through this procedure:

1. Acquire a permit for the message.
2. Write the message into the channel's storage.
3. Make the message available to receivers.

With the `len` method, the length only changes in step 3, but with this method, the capacity already changes in step 1. If the user is using [`Permit`](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Permit.html) to acquire permits before sending messages, then there may be a long amount of time between step 1 and step 2.

The documentation should clearly explain the difference.

Reply:
I've updated the example to explain the relation with `Receiver::len`. Is that sufficient?